src/org/zaproxy/zap/extension/pscanrulesAlpha/StrictTransportSecurityScanner.java

```diff
@@ -47,9 +50,11 @@
 	//and is used to reset browser's settings for HSTS.
 	//If found raise an INFO alert.
 	//Pattern accounts for potential spaces and quotes
-	private static final Pattern BAD_MAX_AGE_PATT = Pattern.compile("max-age\\s*=\\s*\'*\"*\\s*0\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);
+	private static final Pattern BAD_MAX_AGE_PATT = Pattern.compile("\\bmax-age\\s*=\\s*\'*\"*\\s*0\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);
+	//Ensure max-age actually contains a digit
+	private static final Pattern MAX_AGE_PATT = Pattern.compile("\\bmax-age\\s*=\\s*\'*\"*\\s*\\s*\\d+\\s*\"*\'*\\s*", Pattern.CASE_INSENSITIVE);

```

Comment:
Yes, that page has some of the illegal quoting chars.

How is that regex going to be used? It also matches valid quoting. I was going to use the RFC definition to build the regex to just allow the expected chars, if it does not match the regex it would raise the alert.